src/extensions/initializeHitherInterface.ts

```diff
@@ -20,24 +20,26 @@ type HitherCreateJobMessage = {
     kwargs: { [key: string]: any },
     clientJobId: string
 }
-type HitherJobMessage = HitherCancelJobMessage | HitherCreateJobMessage
+export type HitherJobMessage = HitherCancelJobMessage | HitherCreateJobMessage
 
 interface HitherJobCreatedMessageFromServer {
     client_job_id: string
     job_id: string
 }
 
-const initializeHitherInterface = (sendMessage: (msg: HitherJobMessage) => void, baseSha1Url: string) => {
+const initializeHitherInterface = (baseSha1Url: string) => {
     const globalData: {
         // dispatch: Dispatch<RootAction> | null,
         hitherClientJobCache: { [key: string]: ClientHitherJob },
         hitherJobs: { [key: string]: ClientHitherJob }
         runningJobIds: { [key: string]: boolean }
+        sendMessage: (msg: HitherJobMessage) => void
```

Comment:
sendMessage is the function that sends a message to the server - (websocket for web gui and other mechanism for jupyter). Rather than passing `sendMessage` as an arg to `initializeHitherInterface` we need to provide a method to register the `sendMessage` - see below - so that it can be reregistered after a new connection is established.